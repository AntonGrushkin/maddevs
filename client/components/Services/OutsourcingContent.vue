<template>
  <div class="it-outsourcing__outsourcing-content">
    <div class="it-outsourcing__content_wrapper">
      <div class="it-outsourcing__content-wrap">
        <div class="it-outsourcing__title-wrap">
          <h3 class="it-outsourcing__title title default-title">{{ title }}</h3>
          <h4 class="it-outsourcing__sub-title sub-title">{{ subTitle }}</h4>
        </div>
        <img 
          class="it-outsourcing__content-icon" 
          src="@/assets/img/Services/svg/individuals-symbol.svg" 
          alt="individuals-title-symbol"
          v-if="title === 'Individuals'"
        />
        <img 
          class="it-outsourcing__content-icon" 
          src="@/assets/img/Services/svg/teams-symbol.svg" 
          alt="teams-title-symbol"
          v-if="title === 'Teams'"
        />
      </div>
      <p class="it-outsourcing__paragraph paragraph">{{ paragraph }}</p>
    </div>
    <buttonTrigger :buttonInnerText="buttonInnerText" :modalWindowName="modalWindowName" class="it-outsourcing__button-trigger"/>
  </div>
</template>

<script>
import buttonTrigger from '@/components/ui/button-trigger';

export default {
  name: 'OutsourcingContent',
  props: {
    title: {
      type: String,
      default: 'Title'
    },
    subTitle: {
      type: String,
      default: 'Sub Title'
    },
    paragraph: {
      type: String,
      default: 'Paragraph'
    },
    buttonInnerText: {
      type: String,
      default: 'Button Inner Text'
    },
    modalWindowName: {
      type: String,
      default: 'Modal Window Name'
    }
  },
  components: {
    buttonTrigger
  }
};

</script>

<style lang="scss" scoped>
  @import '../../assets/styles/vars';

  .it-outsourcing {
    &__button-trigger {
      width: 100%;
    }

    &__content-icon {
      margin-bottom: 15px;
      position: relative;
      top: -35px;
      opacity: 40%;
    }

    &__outsourcing-content {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      max-width: 47%;
    }

    &__content-wrap {
      display: flex;
    }

    &__title,
    &__sub-title,
    &__paragraph,
    &__button-trigger {
      color: $text-color--grey;
    }

    &__title {
      padding-bottom: 17px;
      font-size: 70px;
      line-height: 92%;
      letter-spacing: -2px;
    }

    &__sub-title {
      font-size: 32px;
      line-height: 110%;
      letter-spacing: -2px;
      margin-bottom: 15px;
    }

    &__paragraph {
      padding-bottom: 46px;
    }
  }

  .individuals {
    .it-outsourcing__content-icon {
      padding-right: 14px;
    }
  }

  .teams,
  .individuals {
    &:hover {
      .it-outsourcing__title,
      .it-outsourcing__paragraph,
      .it-outsourcing__sub-title,
      .it-outsourcing__button-trigger {
        color: $text-color--grey-light;
        transition: 0.3s;
      }

      .it-outsourcing__button-trigger {
        border-color: $border-color--grey-light-outsourcing-section;
      }

      .it-outsourcing__content-icon {
        opacity: 1;
        transition: 0.3s;
      }
    }
  }

  @media only screen and (min-width: 1600px) {
    .it-outsourcing {
      &__sub-title {
        max-width: 85%;
      }
    }
  }

  @media only screen and (max-width: 1280px) {
    .it-outsourcing {
      padding: 170px 0;

      &__main-title {
        padding-bottom: 52px;
      }

      &__title {
        font-size: 50px;
        line-height: 110%;
        letter-spacing: -2px;
        padding-bottom: 25px;
      }

      &__sub-title {
        font-size: 32px;
        line-height: 110%;
        letter-spacing: -1px;
      }

      &__content-icon {
        height: 180px;
      }
    }
  }

  @media only screen and (max-width: 1180px) {
    .it-outsourcing {
      &__title {
        font-size: 50px;
        line-height: 110%;
        letter-spacing: -2px;
      }

      &__content-icon {
        height: 160px;
      }
    }
  }

  @media only screen and (max-width: 1024px) {
    .it-outsourcing {
      button {
        width: initial;
      }

      &__title {
        font-size: 48px;
        padding-bottom: 25px;
      }

      &__sub-title {
        width: initial;
        font-size: 33px;
        line-height: 110%;
        letter-spacing: -1px;
      }

      &__content-icon {
        height: 130px;
        padding-left: 10px;
      }
    }

    .teams,
    .individuals * {
      .it-outsourcing__paragraph {
        width: initial;
      }
    }
  }

  @media only screen and (max-width: 900px) {
    .it-outsourcing {
      &__content-icon {
        height: 115px;
        top: -25px;
      }

      &__title {
        padding-bottom: 20px;
      }
    }
  }

  @media only screen and (max-width: 878px) {
    .it-outsourcing {
      &__content-icon {
        height: 145px;
      }
    }
  }

  @media only screen and (max-width: 768px) {
    .it-outsourcing {
      &__title {
        font-size: 47px;
        padding-bottom: 10px;
      }

      &__sub-title {
        font-size: 32px;
      }

      &__paragraph {
        line-height: 119%;
      }
    }
  }

  @media only screen and (max-width: 580px) {
    .it-outsourcing {
      &__content-icon {
        height: 125px;
        top: -13px;
      }
    }
  }

  @media only screen and (max-width: 520px) {
    .it-outsourcing {
      &__last-character {
        margin-left: -21px;
      }

      &__content-icon {
        height: 115px;
      }
    }
  }

  @media only screen and (max-width: 375px) {
    .it-outsourcing {
      &__main-title {
        padding-bottom: 25px;
      }

      &__content-icon {
        height: 100px;
      }

      &__title {
        font-size: 41px;
        line-height: 99%;
        letter-spacing: -1px;
      }

      &__sub-title {
        font-size: 27px;
        line-height: 124%;
        letter-spacing: -1px;
        margin-bottom: 10px;
      }

      &__paragraph {
        padding-bottom: 34px;
      }
    }
  }

  @media only screen and (max-width: 345px) {
    .it-outsourcing {
      &__content-icon {
        height: 75px;
      }
    }
  }
</style>
