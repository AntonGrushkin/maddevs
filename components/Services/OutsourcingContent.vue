<template>
  <div class="it-outsourcing__outsourcing-content">
    <div class="it-outsourcing__content_wrapper">
      <h1 class="it-outsourcing__title" v-if="title === 'Teams'">
        <span>Tea</span>
        <img 
          class="it-outsourcing__teams-title-icon" 
          src="../../assets/img/Services/svg/teams-sub-title-symbol.svg" 
          alt="teams-sub-title-symbol"
        />
        <span>s</span>
      </h1>
      <h1 class="it-outsourcing__title" v-else>{{ title }}</h1>
      <h2 class="it-outsourcing__sub-title">{{ subTitle }}</h2>
      <p class="it-outsourcing__paragraph">{{ paragraph }}</p>
    </div>
    <buttonTrigger :buttonInnerText="buttonInnerText"/>
  </div>
</template>

<script>
import buttonTrigger from '@/components/ui/button-trigger';

export default {
  name: 'OutsourcingContent',
  props: {
    title: {
      type: String,
      default: 'Title'
    },
    subTitle: {
      type: String,
      default: 'Sub Title'
    },
    paragraph: {
      type: String,
      default: 'Paragraph'
    },
    buttonInnerText: {
      type: String,
      default: 'Button Inner Text'
    }
  },
  components: {
    buttonTrigger
  }
};

</script>

<style lang="scss" scoped>
  .it-outsourcing {
    &__outsourcing-content {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
    }
    
    &__title,
    &__sub-title {
      font-family: 'Hoves-Bold';
    }

    &__title {
      height: 100px;
      padding-bottom: 34px;
      font-size: 90px;
      :last-child {
        margin-left: -33px;
      }
    }

    &__sub-title  {
      padding-bottom: 16px;
      font-size: 50px;
    }

    &__paragraph {
      width: 535px;
      padding-bottom: 46px;
      font-size: 18px;
      font-family: 'Hoves-Regular';
    }

    &__teams-title-icon {
      margin-left: -24px;
      padding: 0 5px;
    }
  }

  .teams * {
    color: #f2f2f2;
    .it-outsourcing__paragraph {
      width: 535px;
    }
  }

  .individuals * {
    color: #96969C;
    .it-outsourcing__paragraph {
      width: 575px;
    }
  }

  @media only screen and (max-width: 1560px) {
    .it-outsourcing {
      button {
        width: 400px;
      }

      &__title {
        font-size: 80px;
      }

      &__sub-title  {
        font-size: 44px;
      }

      &__teams-title-icon {
        height: 77px;
      }
    }
  }

  @media only screen and (max-width: 1390px) {
    .it-outsourcing {
      button {
        width: 380px;
      }

      &__title {
        font-size: 75px;
      }

      &__sub-title  {
        font-size: 40px;
      }
    }
  }

  @media only screen and (max-width: 1270px) {
    .it-outsourcing {
      button {
        width: 360px;
      }

      &__title {
        font-size: 70px;
        padding-bottom: 0;
      }

      &__sub-title  {
        font-size: 35px;
      }

      &__teams-title-icon {
        height: 67px;
      }
    }

    .teams,
    .individuals * {
      .it-outsourcing__paragraph {
        width: 435px;
      }
    }
  }

  @media only screen and (max-width: 1180px) {
    .it-outsourcing {
      button {
        width: 340px;
      }

      &__title {
        font-size: 65px;
      }

      &__sub-title  {
        width: 465px;
        font-size: 30px;
      }
    }

    .teams,
    .individuals * {
      .it-outsourcing__paragraph {
        width: 300px;
      }
    }
  }

  @media only screen and (max-width: 1024px) {
    .it-outsourcing {
      button {
        width: initial;
      }

      &__title {
        font-size: 80px;
      }

      &__sub-title  {
        width: initial;
        font-size: 50px;
      }
    }

    .teams,
    .individuals * {
      .it-outsourcing__paragraph {
        width: initial;
      }
    }

    .individuals {
      padding-top: 104px;
    }
  }

  @media only screen and (max-width: 768px) {
    .it-outsourcing {
      &__title {
        font-size: 65px;
      }

      &__sub-title  {
        font-size: 40px;
      }
    }
  }

  @media only screen and (max-width: 640px) {
    .it-outsourcing {
      &__title {
        font-size: 60px;
      }

      &__sub-title  {
        font-size: 35px;
      }
    }
  }

  @media only screen and (max-width: 520px) {
    .it-outsourcing {
      &__title {
        font-size: 41px;
        :last-child {
          margin-left: -21px;
        }
      }

      &__sub-title  {
        font-size: 27px;
      }

      &__paragraph {
        font-size: 17px;
      }

      &__teams-title-icon {
        height: 45px;
        padding-left: 13px;
      }
    }
  }
</style>