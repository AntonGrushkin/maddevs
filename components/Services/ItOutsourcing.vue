<template>
  <section id="it-outsourcing" class="it-outsourcing main-parent-item">
    <div class="container">
      <div class="it-outsourcing__content-wrapper">
        <h2 class="it-outsourcing__main-title main-title">IT Outsourcing</h2>
        <div class="it-outsourcing__flex-wrapper flex-wrapper">
          <OutsourcingContent 
            :title="teams.title" 
            :subTitle="teams.subTitle" 
            :paragraph="teams.paragraph"
            :buttonInnerText="teams.buttonInnerText"
            class="teams"
          />
          <OutsourcingContent 
            :title="individuals.title" 
            :subTitle="individuals.subTitle" 
            :paragraph="individuals.paragraph"
            :buttonInnerText="individuals.buttonInnerText"
            class="individuals"
          />
        </div>
      </div>
    </div>
  </section>
</template>

<script>
import OutsourcingContent from '@/components/Services/OutsourcingContent';

export default {
  name: 'ItOutsourcing',
  components: {
    OutsourcingContent
  },
  data() {
    return {
      teams: {
        title: 'Teams',
        subTitle: 'Tired of 20 or 30 people not delivering you any features?',
        paragraph: 'How about only dealing with smart heads and capable hands? Enjoy our proactive rockets who will ask you to get them more work.',
        buttonInnerText: 'Get a team of ultra-fast coders'
      },
      individuals: {
        title: 'Individuals',
        subTitle: 'Seeing one, two, five guys working on a minor thing? Ha!',
        paragraph: 'We are NOT going to bring you more people than you actually need. We\'ll just send one or two of our rangers to your frontier, and you\'ll enjoy peace of mind about IT.',
        buttonInnerText: 'Get individual proactive rockets'
      }
    };
  }
};
</script>

<style lang="scss" scoped>
  @import '../../assets/styles/vars';

  .it-outsourcing {
    padding: 240px 0;
    background-color: $bgcolor--black;

    &__main-title {
      padding-bottom: 65px;
      color: $text-color--red;
    }
  }

  @media only screen and (max-width: 1024px) {
    .it-outsourcing {
      &__flex-wrapper {
        flex-direction: column;
      }
    }
  }
</style>
