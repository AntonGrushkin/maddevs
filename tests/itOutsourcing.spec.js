import {
  mount
} from '@vue/test-utils';
import ItOutsourcing from '@/components/Services/ItOutsourcing.vue';
import OutsourcingContent from '@/components/Services/OutsourcingContent.vue';

describe('IT Outsourcing', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(ItOutsourcing);
  });

  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
