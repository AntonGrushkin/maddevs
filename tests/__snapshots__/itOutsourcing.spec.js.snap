// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IT Outsourcing renders correctly 1`] = `
<section
  class="it-outsourcing main-parent-item"
  id="it-outsourcing"
>
  <div
    class="container"
  >
    <div
      class="it-outsourcing__content-wrapper"
    >
      <h2
        class="it-outsourcing__main-title main-title"
      >
        IT Outsourcing
      </h2>
       
      <div
        class="it-outsourcing__flex-wrapper flex-wrapper"
      >
        <div
          class="it-outsourcing__outsourcing-content teams"
        >
          <div
            class="it-outsourcing__content_wrapper"
          >
            <div
              class="it-outsourcing__content-wrap"
            >
              <div
                class="it-outsourcing__title-wrap"
              >
                <h3
                  class="it-outsourcing__title title default-title"
                >
                  Teams
                </h3>
                 
                <h4
                  class="it-outsourcing__sub-title sub-title"
                >
                  Tired of 20 or 30 people not delivering you any features?
                </h4>
              </div>
               
              <!---->
               
              <img
                alt="teams-title-symbol"
                class="it-outsourcing__content-icon"
                src="@/assets/img/Services/svg/teams-symbol.svg"
              />
            </div>
             
            <p
              class="it-outsourcing__paragraph paragraph"
            >
              How about only dealing with smart heads and capable hands? Enjoy our proactive rockets who will ask you to get them more work.
            </p>
          </div>
           
          <button
            class="ui-button-modal-trigger it-outsourcing__ui-button-modal-trigger"
          >
            
  Get a team of ultra-fast coders

          </button>
        </div>
         
        <div
          class="it-outsourcing__outsourcing-content individuals"
        >
          <div
            class="it-outsourcing__content_wrapper"
          >
            <div
              class="it-outsourcing__content-wrap"
            >
              <div
                class="it-outsourcing__title-wrap"
              >
                <h3
                  class="it-outsourcing__title title default-title"
                >
                  Individuals
                </h3>
                 
                <h4
                  class="it-outsourcing__sub-title sub-title"
                >
                  Seeing one, two, five guys working on a minor thing? Ha!
                </h4>
              </div>
               
              <img
                alt="individuals-title-symbol"
                class="it-outsourcing__content-icon"
                src="@/assets/img/Services/svg/individuals-symbol.svg"
              />
               
              <!---->
            </div>
             
            <p
              class="it-outsourcing__paragraph paragraph"
            >
              We are NOT going to bring you more people than you actually need. We'll just send one or two of our rangers to your frontier, and you'll enjoy peace of mind about IT.
            </p>
          </div>
           
          <button
            class="ui-button-modal-trigger it-outsourcing__ui-button-modal-trigger"
          >
            
  Get individual proactive rockets

          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
